Sources/NIO/ConvenienceOptionSupport.swift

```diff
@@ -0,0 +1,178 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2020 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+// MARK:  Universal Client Bootstrap
+extension NIOClientTCPBootstrapProtocol {
+    /// Apply any understood shorthand options to the bootstrap, removing them from the set of options if they are consumed.
+    /// - parameters:
+    ///     - options:  The options to try applying - the options applied should be consumed from here.
+    /// - returns: The updated bootstrap with and options applied.
+    public func _applyOptions(_ options: inout NIOTCPShorthandOptions) -> Self {
+        // Default is to consume no options and not update self.
+        return self
+    }
+}
+
+extension NIOClientTCPBootstrap {
+    /// Specifies some `ChannelOption`s to be applied to the channel.
+    /// - See: channelOption
+    /// - Parameter options: Set of shorthand options to apply.
+    /// - Returns: The updated bootstrap (`self` being mutated)
+    public func options(_ options: NIOTCPShorthandOptions) -> NIOClientTCPBootstrap {
```

Comment:
@Davidde94 / @glbrntt / @Lukasa this function name needs to be bike-shed. It's a new recommended-to-use public API. I think it needs to have the word `channel` in there.

Reply:
`channelOptions` is indeed not good imho because it's confusing to have `channelOption` and `channelOptions` being totally different things.

But we could have `channelConvenienceOptions` or something like that.

Reply:
Some thoughts:

- `generalChannelOptions`
- `genericChannelOptions`
- `universalChannelOptions`
- `shorthandChannelOptions`

Reply:
I think general, generic and universal are all words used elsewhere to mean something else and don't feel like the right fit.
Shorthand or Convenience both feel fine.  Happy with either.